conda_forge_tick/update_recipe/v2/jinja/jinja.py

```diff
@@ -0,0 +1,94 @@
+from __future__ import annotations
+
+from typing import TypedDict
+
+import jinja2
+from jinja2.sandbox import SandboxedEnvironment
+
+from conda_forge_tick.update_recipe.v2.jinja.filters import (
+    _bool,
+    _split,
+    _version_to_build_string,
+)
+from conda_forge_tick.update_recipe.v2.jinja.objects import (
+    _stub_compatible_pin,
+    _stub_is_linux,
+    _stub_is_unix,
+    _stub_is_win,
+    _stub_match,
+    _stub_subpackage_pin,
+    _StubEnv,
+)
+from conda_forge_tick.update_recipe.v2.jinja.utils import _MissingUndefined
+
+# from conda_forge_tick.update_recipe.v2.loader import load_yaml
+
+
+class RecipeWithContext(TypedDict, total=False):
+    context: dict[str, str]
+
+
+def jinja_env() -> SandboxedEnvironment:
+    """
+    Create a `rattler-build` specific Jinja2 environment with modified syntax.
+    Target platform, build platform, and mpi are set to linux-64 by default.
+    """
+    env = SandboxedEnvironment(
+        variable_start_string="${{",
+        variable_end_string="}}",
+        trim_blocks=True,
+        lstrip_blocks=True,
+        autoescape=jinja2.select_autoescape(default_for_string=False),
+        undefined=_MissingUndefined,
+    )
```

Comment:
Can you add a comment on these options, why they are set, and how that relates to the supported features for the v2 recipe?

Also, will this parser fail on `if` conditions, `for` loops, or other non-supported jinja2 filters / syntax?

Reply:
Is there a way we can make it fail and output an error?